### PR TITLE
Unix.open_process_args_*: look up program in PATH

### LINKED ISSUE #10083: Usefulness of Unix.open_process_args* functions is limited by them not respecting PATH environment variable
As an alternative to the other `Unix.open_process*` functions (that take a single string and rely on `Filename.quote_command`), these `Unix.open_process_args*` functions introduced in 4.08 are nice. But crucially they don't currently respect the `PATH` environment variable, limiting their usefulness.

e.g. I'd expect the following to work, since `/usr/bin` is mentioned in my `PATH` and the executable `/usr/bin/uname` exists.

```ocaml
let ic, oc = Unix.open_process_args "uname" [| "uname"; "-a" |]
let status = Unix.close_process (ic, oc)
```

But `status` is `Unix.WEXITED 127` because the executable couldn't be found.

I think this is beacause the implementation is using the `execv` syscall rather than `execvp`.

### PR DESCRIPTION
**Pull request by nojb**
Fixes #10083, see the discussion in #10057, especially https://github.com/ocaml/ocaml/pull/10057#issuecomment-735073429 and https://github.com/ocaml/ocaml/pull/10057#issuecomment-735254729

cc @xavierleroy

### CODE REVIEW

File: otherlibs/unix/unix.mli
Comment (gasche) [inline otherlibs/unix/unix.mli]: I think we would need a `@since` marker for the PATH lookup, or at least a clear indication in the documentation that older versions of OCaml may not do it.

File: otherlibs/unix/unix.ml
Comment (gasche) [inline otherlibs/unix/unix.ml:920]: if `spawn` is not exported, we should make `usepath` a labelled argument instead of using comments to alleviate the problems of non-labelled boolean arguments.

(Looking at the code, I see that `spawn` is an external. Can external have labels too?)

File: Changes
Comment (frou) [inline Changes]: The glob only matches 3 of the 4 functions because of the trailing underscore 😈

Comment (xavierleroy) [reply Changes]: This is a breaking change, strictly speaking, so it should be listed with a `*` instead of a `-`.

Why is this breaking?  Assume the current directory contains an executable named `foo.exe`, and `PATH` does not contain `.`, like it should.  `Unix.open_process_args "foo.exe" ...` succeeds with the current implementation and fails with the new implementation.

File: otherlibs/unix/unix.ml
Comment (xavierleroy) [reply otherlibs/unix/unix.ml:920]: Externals can have labeled and even optional arguments.  There are examples elsewhere in this `unix.ml` file.

This said, `spawn` is not exported, and is called  only twice in `unix.ml`, so it doesn't need to have a bulletproof interface.

File: Changes
Comment (nojb) [reply Changes]: Good point, I amended the `Changes` entry and the docstring for the function.

File: otherlibs/unix/unix.ml
Comment (nojb) [reply otherlibs/unix/unix.ml:920]: OK, I'm not adding a label as it seems a bit overkill in this case...

File: otherlibs/unix/unix.mli
Comment (nojb) [reply otherlibs/unix/unix.mli]: I added a clarification about this to the docstring, but without using a `@since` marker since it wasn't clear to me how to specify more than one such marker.

File: Changes
Comment (gasche) [inline Changes]:
```suggestion
  Unix. Note that previously the program was interpreted relative to the current
```

File: otherlibs/unix/unix.mli
Comment (gasche) [inline otherlibs/unix/unix.mli]: Could you break a line before this new sentence, so that the behavior changes catches the eye?

**Review by gasche — APPROVED** [review]
I am in favor of the change, which improves compatibility between the different platforms and seems more consistent with what our users actually want.

File: Changes
Comment (nojb) [reply Changes]: Thanks, fixed.

File: otherlibs/unix/unix.mli
Comment (nojb) [reply otherlibs/unix/unix.mli]: Done.

**Review by gasche — APPROVED** [review]
This is an approval stamp.

Before we merge, do we need to do anything specific to the fact that we are proposing a change to the sdlib that is "a bugfix" or "a breaking change" according to who you ask?

I was wondering if we needed more opinions on this; let me cc the @ocaml/caml-devel group, in case people want to chime in. If there is no negative feedback for about a week, please feel free to remind me to merge.

**Comment by nojb** [comment]
> I was wondering if we needed more opinions on this; let me cc the @ocaml/caml-devel group, in case people want to chime in. If there is no negative feedback for about a week, please feel free to remind me to merge.

Sounds good, thanks!

File: otherlibs/unix/unix.mli
Comment (xavierleroy) [inline otherlibs/unix/unix.mli]: Elsewhere in unix.mli we write "The executable file is searched in the path".  (4 occurrences.)  

Also, you first talk about "the command" and then "the program".  For me, "the command" is a shell command line, as used in `open_process_in` for instance.  "The program" is better.

Also, "the program was interpreted relative to the current directory" is still true: the path is not searched for programs like `./foo` or `foo/bar`.

**Review by xavierleroy — COMMENTED** [review]
I have zero problems with the change of semantics.  Yes, technically it is a breaking change, so that's why it must be starred in the change log.  But the previous behavior was an oversight, and the only way to do something useful with the previous behavior was to use absolute paths `/bin/ls`, in which case the new behavior is the same.  So, I don't think we have an issue here.

Nonetheless I have hairs to split about the documentation strings, see below.

File: otherlibs/unix/unix.mli
Comment (nojb) [reply otherlibs/unix/unix.mli]: Thanks, I pushed revised docs for the `open_process_args*` functions to make them more uniform with the rest of the docstrings in this file (also added a line about the environment passed to the new process, which was previously missing). Let me know what you think!

**Review by xavierleroy — APPROVED** [review]
The new documentation is just fine (modulo a typo that I fixed by a direct push), thank you very much!

I'm so happy that I'll merge right now.

**Pull request merged**
